### PR TITLE
feat(unified-share-modal): Support avatars in contacts dropdown

### PR DESCRIPTION
**Pull request by gowong**

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @gowong, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

**Comment by gowong** [comment]
CLA signed

File: src/features/unified-share-modal/flowTypes.js [hunk 1]
```diff
@@ -36,7 +36,12 @@ export type accessLevelsDisabledReasonType = {
 };
 
 export type contactType = {
+    avatarUrls?: {
```
Comment (naeims) [inline src/features/unified-share-modal/flowTypes.js]: This is now called `avatarURLs`, note the casing.

**Comment by booc0mtaco** [comment]
This should be re-posted, since some of the changes were extracted into smaller, more focused PRs.

**Comment by booc0mtaco** [comment]
- Avatar changes implemented via #1951 
- LabelPill changes implemented via #1950

File: src/components/avatar/Avatar.js.flow [hunk 2]
```diff
@@ -7,7 +7,7 @@ import UnknownUserAvatar from '../../icons/avatars/UnknownUserAvatar';
 
 import './Avatar.scss';
 
-const SIZES = { large: true };
```
Comment (booc0mtaco) [inline src/components/avatar/Avatar.js.flow:10]: RE: #2149

File: src/components/avatar/Avatar.scss [hunk 3]
```diff
@@ -7,6 +7,8 @@
 .avatar {
     position: relative;
     display: inline-block;
```
Comment (booc0mtaco) [inline src/components/avatar/Avatar.scss:9]: RE: #2149

File: src/components/avatar/Avatar.stories.tsx [hunk 4]
```diff
@@ -4,6 +4,10 @@ import notes from './Avatar.stories.md';
 
 export const regular = () => <Avatar id={1} name="Aaron Levie" />;
 
+export const small = () => <Avatar id={1} name="Aaron Levie" size="small" />;
```
Comment (booc0mtaco) [inline src/components/avatar/Avatar.stories.tsx:7]: RE: #2149

File: src/components/avatar/Avatar.tsx [hunk 5]
```diff
@@ -6,7 +6,7 @@ import UnknownUserAvatar from './UnknownUserAvatar';
 
 import './Avatar.scss';
 
-const SIZES = { large: true };
```
Comment (booc0mtaco) [inline src/components/avatar/Avatar.tsx:9]: RE: #2149

File: src/components/avatar/__tests__/Avatar.stories.test.tsx [hunk 6]
```diff
@@ -1,6 +1,8 @@
 describe('components/avatar/Avatar', () => {
     test.each([
         ['components-avatar--regular'],
+        ['components-avatar--small'],
```
Comment (booc0mtaco) [inline src/components/avatar/__tests__/Avatar.stories.test.tsx:4]: RE: #2149

File: src/components/avatar/__tests__/Avatar.test.tsx [hunk 7]
```diff
@@ -12,7 +12,12 @@ describe('components/avatar/Avatar', () => {
         expect(wrapper.is('span.avatar.test-avatar')).toBe(true);
     });
 
-    test('should add size class based on prop', () => {
+    test('should add small size class based on prop', () => {
```
Comment (booc0mtaco) [inline src/components/avatar/__tests__/Avatar.test.tsx:15]: RE: #2149

File: src/components/contact-datalist-item/ContactDatalistItem.js [hunk 8]
```diff
@@ -1,19 +1,26 @@
 // @flow
 import * as React from 'react';
 
+import Avatar from '../avatar';
```
Comment (booc0mtaco) [inline src/components/contact-datalist-item/ContactDatalistItem.js:4]: RE: #2150

File: src/components/contact-datalist-item/ContactDatalistItem.scss [hunk 9]
```diff
@@ -1,12 +1,31 @@
 @import '../../styles/variables';
 
+.contact-data-list-item {
```
Comment (booc0mtaco) [inline src/components/contact-datalist-item/ContactDatalistItem.scss:3]: RE: #2150

File: src/components/contact-datalist-item/__tests__/ContactDatalistItem.test.js [hunk 10]
```diff
@@ -3,9 +3,10 @@ import React from 'react';
 import ContactDatalistItem from '../ContactDatalistItem';
 
 describe('components/contact-datalist-item/ContactDatalistItem', () => {
-    test('should render a DatalistItem with name and subtitle', () => {
-        const wrapper = shallow(<ContactDatalistItem name="name" subtitle="subtitle" />);
+    test('should render a DatalistItem with avatar, name, and subtitle', () => {
```
Comment (booc0mtaco) [inline src/components/contact-datalist-item/__tests__/ContactDatalistItem.test.js:6]: RE: #2150

File: src/components/label-pill/LabelPill.scss [hunk 11]
```diff
@@ -1,51 +1,70 @@
 @import '../../styles/variables';
 
 .bdl-LabelPill {
-    padding: 3px;
```
Comment (booc0mtaco) [inline src/components/label-pill/LabelPill.scss:4]: LabelPill changes were integrated months before

**Comment by booc0mtaco** [comment]
@gowong @alexkrolick I think that the bulk of this change is now merged to varying degrees. May be best to drop this PR entirely (unless there are bits that can and should be merged in, but arent yet). Please re-open if so!

**Comment by alexkrolick** [comment]
Cool, glad it got there eventually

**Pull request closed without merging**